packages/plugins/preview/src/lib/plugin/NinetailedPreviewPlugin.ts

```diff
@@ -700,6 +696,56 @@ export class NinetailedPreviewPlugin
     return `${experienceId}:${key}`;
   }
 
+  /**
+   * Ensure inline variables for a given experience reflect the selected variant
+   * without triggering multiple onChange calls. Keeps custom flags in sync with forced variants.
+   */
+  private applyInlineVariableOverridesForExperience(
+    experience: ExperienceConfiguration,
+    variantIndex: number
+  ): boolean {
+    let hasChanges = false;
+    const overrides = { ...this.variableOverwrites };
+
+    experience.components.forEach((component) => {
+      if (component.type !== ComponentTypeEnum.InlineVariable) return;
+
+      const value =
+        variantIndex === 0
+          ? component.baseline.value
+          : component.variants[variantIndex - 1]?.value ??
+            component.baseline.value;
+
+      const overrideKey = this.getOverrideKey(experience.id, component.key);
+      const previousOverride = overrides[overrideKey];
+      const nextChange: Change = {
+        type: ChangeTypes.Variable,
+        key: component.key,
+        value,
+        meta: {
+          experienceId: experience.id,
+          variantIndex,
+        },
+      };
+
+      if (
+        previousOverride &&
+        isEqual(previousOverride.value, value) &&
+        previousOverride.meta?.variantIndex === variantIndex
+      )
+        return;
+
+      overrides[overrideKey] = nextChange;
+      hasChanges = true;
+    });
+
+    if (hasChanges) {
+      this.variableOverwrites = overrides;
+    }
+
+    return hasChanges;
```

Comment:
Why do we need to return? The return value is not consumed anywhere. Consequently, we can get rid of the let statement.